File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0 , right = heights.length -1 ;
        int ans  = 0 ;
        while(left<right){
            int area  = Math.min(heights[left],heights[right])* (right -left);
            ans = Math.max(ans,area);
            if(heights[left]<heights[right])left++;
            else if(heights[right]<heights[left])right--;
            else {
                left++;
                right--;
            }

        }
        return ans  ;
    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isValidBST(root) {
        return check(root, -Infinity, Infinity)
    }
}

function check(root, min , max  ){
 if(!root) return true 
 if(root.val <= min || root.val >=max) return false 

 return (
    check(root.left , min , root.val) && check(root.right , root.val , max )
 )
}
